source/angular-time-picker.js

```diff
@@ -22,40 +19,47 @@ angular.module( "vokal.timePicker", [] )
 
                     ngModelController.$setValidity( "time", !isNaN( timeData.getTime() ) );
 
-                    return timeData;
+                    return attrs.pickerType === "string" ?
+                        $filter( "date" )( timeData, attrs.timePicker || "h:mm a" ) : timeData;
                 } );
 
                 // Convert data from model to view format and validate
                 ngModelController.$formatters.push( function( data )
                 {
-                    if( data )
+                    var validTime, timeData = data;
+
+                    if( timeData )
                     {
-                        ngModelController.$setValidity( "time", data.getTime && !isNaN( data.getTime() ) );
+                        if( typeof timeData.getTime !== "function" )
+                        {
+                            timeData = new Date( "1/1/1990 " + timeData );
+                        }
+
+                        validTime = !isNaN( timeData.getTime() );
+
+                        ngModelController.$setValidity( "time", validTime );
                     }
 
-                    return data ? $filter( "date" )( data, "shortTime" ) : "";
+                    return validTime ? $filter( "date" )( timeData, attrs.timePicker || "h:mm a" ) : data;
                 } );
 
                 // Initialize
                 scope.times = [];
                 scope.showTimepicker = false;
-                var options  = attrs.timePickerOptions ? JSON.parse( attrs.timePickerOptions ) : {};
-                var interval = options.interval || 60;
-                var hour, minute, apm;
+                var interval = attrs.pickerInterval ? parseInt( attrs.pickerInterval, 10 ) : 60;
+                var workingTime, minute, formattedTime;
 
                 // Build array of time objects by interval
                 for( var i = 0; i < 24; i++ )
                 {
                     for( var k = 0; k < 60; k += interval )
                     {
-                        hour   = i > 12 ? i - 12 : i;
-                        hour   = hour === 0 ? hour + 12 : hour;
-                        minute = k < 10 ? "0" + k : k;
-                        apm    = i > 11 ? "PM" : "AM";
-                        scope.times.push( { display: hour + ":" + minute + " " + apm, value: i + ":" + minute } );
+                        minute        = k < 10 ? "0" + k : k;
+                        workingTime   = new Date( "1/1/1990 " + i + ":" + minute );
```

Comment:
We use this new Date function in a few places, thoughts on creating a global function out of it?